### ISSUE TITLE
Adding HTTP request headers doesn't work (adafruit_esp32spi_requests.py)

### ISSUE THREAD

**Issue by anecdata** [adafruit/Adafruit_CircuitPython_ESP32SPI#24]
There is a facility for any HTTP request (HEAD, GET, POST, etc.) to have optional user-specified HTTP headers added. For example, 'Http-X-MyCustomHeader: BlahBlahBlah'.
```
def request(method, url, data=None, json=None, headers=None, stream=False):
...'headers' is optional HTTP headers sent along.
```
Current version pieces together the header components to write the header to the socket based on user-specified dict containing one or more key-value pairs describing the header name and value, but there's a runtime error when trying to process the dictionary I believe due to unencoded string elements of the headers: 
```
  File "/lib/adafruit_esp32spi/adafruit_esp32spi_wifimanager.py", line 110, in get
  File "/lib/adafruit_esp32spi/adafruit_esp32spi_requests.py", line 235, in get
  File "/lib/adafruit_esp32spi/adafruit_esp32spi_requests.py", line 221, in request
  File "/lib/adafruit_esp32spi/adafruit_esp32spi_requests.py", line 176, in request
  File "/lib/adafruit_esp32spi/adafruit_esp32spi_socket.py", line 82, in write
  File "/lib/adafruit_esp32spi/adafruit_esp32spi.py", line 531, in socket_write
  File "/lib/adafruit_esp32spi/adafruit_esp32spi.py", line 298, in _send_command_get_response
  File "/lib/adafruit_esp32spi/adafruit_esp32spi.py", line 210, in _send_command
TypeError: can't convert str to int
```
To fix, I _think_ it's just a matter of changing lines 176 & 178 in adafruit_esp32spi_requests.py to encode the strings into (default ASCII) bytestrings, changing from:
```
        for k in headers:
            sock.write(k)
            sock.write(b": ")
            sock.write(headers[k])
            sock.write(b"\r\n")
```
to:
```
        for k in headers:
            sock.write(k.encode())
            sock.write(b": ")
            sock.write(headers[k].encode())
            sock.write(b"\r\n")
```
I tested it in a basic scenario, and the program runs, and I receive the custom request headers on my server. But I'd really appreciate more seasoned Python eyes looking at this.

**Comment by makermelissa** [comment]
Hi, do you want to submit a PR and then some more “seasoned” eyes could take a look at that during the review? 🙂

**Comment by anecdata** [comment]
Sure, I'll give it a shot. There's a first time for everything. I'll do some reading on the process and see what I need to set up to do that.

**Comment by makermelissa** [comment]
https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

**Comment by brentru** [comment]
Fixed in release 1.3.4! Closing...
